cli/internal/api/api.go

```diff
@@ -93,15 +95,22 @@ func (c *Client) StreamLogs(ctx context.Context, w io.Writer, workflowName strin
 		return fmt.Errorf("received unexpected status code: %d", resp.StatusCode)
 	}
 
-	_, err = io.Copy(w, resp.Body)
+	// discard reader bytes already logged
+	discardedWriter := &bytes.Buffer{}
+	if _, err := io.CopyN(discardedWriter, resp.Body, loggedBytes); err != nil {
+		return err
+	}
+	loggedBytes, err = io.Copy(w, resp.Body)
```

Comment:
is it hard to cover this with tests? eg. mock Reader to return the INTERNAL_ERROR n-times with a test case that succeeds and another that exceeds max retries?